### PR TITLE
API SilverStripe 4 compatibility

### LINKED ISSUE #17: SilverStripe 4 compatibility
Any plans for SS4 compatibility?

### PR DESCRIPTION
**Pull request by robbieaverill**
This pull request requires `master` to be branched off to `1` for existing version support, then #45 merged.

---

This PR:

* Implements namespaces for the module and the core packages
* Adds [upgrader](https://github.com/silverstripe/silverstripe-upgrader) mapping
* Updates Travis configuration to use "new" style
* Replace DebugBarLogWriter with Monolog configuration and a formatter
* Replace DebugBarRequestFilter with an [HTTP middleware](https://docs.silverstripe.org/en/4/developer_guides/controllers/middlewares/) implementation
* Removes SQLFormatter in favour of installing with Composer (unstable version, as the module is fairly unsupported these days - this could be brought back to a stable tag if necessary)

Resolves #17

### CODE REVIEW

File: code/Middleware/DebugBarMiddleware.php [hunk 1]
```diff
@@ -1,53 +1,63 @@
...
+     * @param HTTPRequest $request
      */
-    public function preRequest(SS_HTTPRequest $request, Session $session, DataModel $model)
+    protected function beforeRequest(HTTPRequest $request)
     {
-        DebugBar::withDebugBar(function (DebugBar\DebugBar $debugbar) {
-            /* @var $timeData DebugBar\DataCollector\TimeDataCollector */
-            $timeData = $debugbar['time'];
+        DebugBar::withDebugBar(function (\DebugBar\DebugBar $debugbar) {
+            /** @var DebugBar\DataCollector\TimeDataCollector $timeData */
```
Comment (fspringveldt) [inline code/Middleware/DebugBarMiddleware.php:29]: Minor: should be \LeKoala\DebugBar\Collector\timeDataCollector

Comment (robbieaverill) [reply code/Middleware/DebugBarMiddleware.php:29]: This is correct or at least valid as it's the parent class for our extended class

Comment (fspringveldt) [reply code/Middleware/DebugBarMiddleware.php:29]: Yes you're right.

**Review by fspringveldt — APPROVED** [review]

**Comment by robbieaverill** [comment]
Self merging as approved by @fspringveldt

**Pull request merged**